objcgen/driver.cs

```diff
@@ -207,11 +207,44 @@ public void SetCompilationTarget (string value)
 		public TargetLanguage TargetLanguage { get; private set; } = TargetLanguage.ObjectiveC;
 		public CompilationTarget CompilationTarget { get; set; } = CompilationTarget.SharedLibrary;
 
+		public string PlatformSdkDirectory {
+			get {
+				switch (Platform) {
+				case Platform.iOS:
+					return "/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS";
+				case Platform.tvOS:
+					return "/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.TVOS";
+				case Platform.watchOS:
+					return "/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.WatchOS";
+				case Platform.macOS:
+					throw new NotImplementedException ("mac platform sdk"); // need to know full/mobile
```

Comment:
Yeah, the `Platform` update comes with XM support as well, so I think it's better in a new PR.